pkgs/games/shipwright/default.nix

```
@@ -0,0 +1,203 @@
+    hash = "sha256-nDjaFrlBYXgxBW4BvsEI1uJVvXHgNwsro44B2Td85bI=";
+    fetchSubmodules = true;
+  };
+  rev = src.rev;
+
+  enableParallelBuilding = true;
+
+  # Copied from the building guide
+  cmakeFlags = [
+    "DEBUG=0"
+    "OPTFLAGS=-O2"
+  ];
+
+  # Linking fails without this
+  hardeningDisable = [ "all" ];
+
+  # Many warnings are being treated as errors
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-error"
+  ];
+
+  nativeBuildInputs = [ python3 llvmPackages.bintools copyDesktopItems makeWrapper cmake ninja pkg-config lsb-release clang imagemagick ];
+
+  buildInputs = [ libX11 libXrandr libXinerama libXcursor libXi libXext glew SDL2 SDL2_net libpulseaudio libpng bzip2 StormLib ];
+
+  postPatch = ''
+    # This script attempts to call git to retrieve the commit hash,
+    # so we have to manually patch it in instead.
+    substituteInPlace ZAPDTR/ZAPD/genbuildinfo.py \
+      --replace "label = subprocess" "#label = subprocess" \
+      --replace "+ label +" "+ \"${builtins.substring 0 7 rev}\" +"
+
+    # LIBULTRASHIP: Patch the localdir function to point to dir in user's home.
+    substituteInPlace libultraship/src/core/Window.cpp \
+      --replace "return \".\";" "return ((std::string) std::getenv(\"HOME\")) + \"/.local/share/shipofharkinian\";" \
+
+    # SHIPWRIGHT:  to point to dir in user's home.
+    substituteInPlace soh/soh/OTRGlobals.cpp \
+      --replace "Ship::Window::GetPathRelativeToAppDirectory(\"oot-mq.otr\");" "(\"./oot-mq.otr\");" \
+      --replace "Ship::Window::GetPathRelativeToAppDirectory(\"oot.otr\");" "(\"./oot.otr\");" \
+      --replace "Ship::Window::GetPathRelativeToAppDirectory(\"mods\");" \
+        "((std::string) std::getenv(\"HOME\")) + \"/.local/share/shipofharkinian/mods\";"
+  '';
+
+  buildPhase = ''
+    # Generate Ninja project
+    cmake -H. -Bbuild-cmake -GNinja -DCMAKE_BUILD_TYPE:STRING=Release
+
+    # Extract assets & generate OTR
+    ${lib.optionalString enableOotRom "ln -s '${ootRom}' OTRExporter/oot.z64"}
+    ${lib.optionalString enableOotMqRom "ln -s '${ootMqRom}' OTRExporter/oot-mq.z64"}
+    cmake --build build-cmake --target ExtractAssets -v
+
+    # Make sure it actually made the OTR(s)
+    ${lib.optionalString enableOotRom "if [ ! -e OTRExporter/oot.z64 ]; then exit 1; fi"}
+    ${lib.optionalString enableOotMqRom "if [ ! -e OTRExporter/oot-mq.z64 ]; then exit 1; fi"}
+
+    # Build!
+    cmake --build build-cmake --config Release
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    cd build-cmake/soh
+    mkdir -p $out/{bin,lib,share/pixmaps}
+    cp soh.elf $out/lib
+    ${lib.optionalString enableOotRom "cp oot.otr $out/lib"}
+    ${lib.optionalString enableOotMqRom "cp oot-mq.otr $out/lib"}
+    cp ../../OTRExporter/assets/ship_of_harkinian/icons/gSohIcon.png $out/share/pixmaps/soh.png
+    ls -R ../.. | grep gSohIcon
+    # oot.otr needs to be in the same directory as soh itself
+    makeWrapper $out/lib/soh.elf $out/bin/soh \
+      --chdir "$out/lib"
```

Using `makeWrapper` seems a bit overkill, this could just be a symlink. It maybe also makes sense to use `$out/share` instead for the blobs?
```suggestion
    # oot.otr needs to be in the same directory as the executable
    ln -s $out/lib/soh.elf $out/bin/soh
```